Add checkstyle and refactor code

S1mm1ssim opened the pull request:
- Add checkstyle plugin with checkstyle.xml
- Make EntityNotFoundException extend RuntimeException and not regular Exception
- Remove unused imports
- In DAO classes make table names constants and refactor queries using String.format
- In DAO classes refactor ResultSet.getX use String constants and not plain strings
- Remove lombok onConstructor_ = {@Autowired}

S1mm1ssim committed:
Make EntityNotFoundException.java extend RuntimeException

S1mm1ssim committed:
Add maven-checkstyle plugin

S1mm1ssim committed:
Remove unused import

S1mm1ssim committed:
Make table names and their column names constant

S1mm1ssim committed:
Fix DataSource.java configuration

Use properties file to set up config

S1mm1ssim committed:
Make Position table name and its column names constant

Versach commented on src/main/java/com/modsensoftware/marketplace/controller/CategoryController.java:
```diff
@@ -40,8 +38,7 @@ public ResponseEntity<List<Category>> getAllCategories() {
     @GetMapping(value = "/{id}", produces = {"application/json"})
-    public ResponseEntity<Category> getCategoryById(@PathVariable(name = "id") Long id)
-            throws EntityNotFoundException {
+    public ResponseEntity<Category> getCategoryById(@PathVariable(name = "id") Long id) {
```
все еще не понимаю зачем тебе ResponseEntity тут

S1mm1ssim committed:
Remove ResponseEntity from controller classes and use @ResponseStatus instead

S1mm1ssim committed:
Bugfix: setting CategoryDto parentId value to null did not have effect on already persisted category object

- Add nullParent property in CategoryDto to meet client's intentions on making parentId null
- Implement CategoryMapper without Mapstruct
- Fix null checks in CategoryDao.update()

S1mm1ssim committed:
Remove setIfNotNull in CategoryDao.update() when setting parentId due to NPE if parent was null previously
